src/parseTools.js

```diff
@@ -29,6 +31,7 @@ function processMacros(text) {
 // Param filenameHint can be passed as a description to identify the file that is being processed, used
 // to locate errors for reporting and for html files to stop expansion between <style> and </style>.
 function preprocess(text, filenameHint) {
+  currentlyParsedFilename = filenameHint;
```

Comment:
please `null` out this global var on exit from this function, to avoid stale data being confusing.